core/arch/x86_64/mm/core_mmu.c

```diff
@@ -215,20 +213,16 @@ static void mmu_unlock(uint32_t exceptions)
 	cpu_spin_unlock_xrestore(&mmu_spinlock, exceptions);
 }
 
-static void clear_user_map(void)
+static void clear_user_map(short int thread_id)
 {
-	int row = ARRAY_SIZE(g_thread_user_ta_pte);
-	int i;
-
-	memset((void *)&g_thread_user_ta_pd[thread_get_id()][0],
-						0, sizeof(g_thread_user_ta_pd[thread_get_id()]));
+	memset((void *)&g_thread_user_ta_pd[thread_id][0],
+						0, sizeof(g_thread_user_ta_pd[thread_id]));
 
-	for (i = 0 ; i < row ; i++)
-		memset((void *)&g_thread_user_ta_pte[thread_get_id()][i][0], 0,
-				sizeof(g_thread_user_ta_pte[thread_get_id()][0]));
+	memset((void *)&g_thread_user_ta_pte[thread_id][0][0], 0,
+			(sizeof(uint64_t) * NO_OF_USER_PT_TABLES * NO_OF_PT_ENTRIES));
 
-	pd_user_counter[thread_get_id()] = 0;
-	pt_user_index[thread_get_id()] = 0;
+	pd_user_counter[thread_id] = 0;
```

Comment:
Do we need pd_user_counter? We can assume this is always 0, which can cover 1G user space. That's enough for now.